File: week07/32/241018.cpp
class Solution {
public:
    bool searchMatrix(vector<vector<int>>& matrix, int target) {
        int row=0, col = matrix[0].size()-1;
        while (col >= 0 && row <= matrix.size()-1) {
            if (matrix[row][col] == target) {
                return true;
            } else if (matrix[row][col] < target) {
                row++;
            } else {
                col--;
            }
        }
        return false;
    }
};

File: week07/32/241018.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        row = 0
        col = len(matrix[0])-1

        while (col >= 0 and row <= len(matrix)-1):
            if (matrix[row][col] == target):
                return True
            elif (matrix[row][col] < target):
                row += 1
            else:
                col -= 1

        return False
        